packages/vats/src/vat-bank.js

```diff
@@ -171,6 +171,138 @@ export function buildRootObject(_vatPowers) {
 
       /** @type {Store<string, Bank>} */
       const addressToBank = makeStore('address');
+
+      /** @type {NotifierRecord<string[]>} */
+      const {
+        notifier: accountsNotifier,
+        updater: accountsUpdater,
+      } = makeNotifierKit([...addressToBank.keys()]);
+
+      /**
+       * Create a new personal bank interface for a given address.
+       *
+       * @param {string} address lower-level bank account address
+       * @returns {Bank}
+       */
+      const getBankForAddress = address => {
+        assert.typeof(address, 'string');
+        if (addressToBank.has(address)) {
+          return addressToBank.get(address);
+        }
+
+        /** @type {WeakStore<Brand, VirtualPurse>} */
+        const brandToVPurse = makeWeakStore('brand');
+
+        /** @type {Bank} */
+        const bank = Far('bank', {
+          getAssetSubscription() {
+            return harden(assetSubscription);
+          },
+          async getPurse(brand) {
+            if (brandToVPurse.has(brand)) {
+              return brandToVPurse.get(brand);
+            }
+
+            const assetRecord = brandToAssetRecord.get(brand);
+            if (!bankCall) {
+              // Just emulate with a real purse.
+              const purse = await E(assetRecord.issuer).makeEmptyPurse();
+              brandToVPurse.init(brand, purse);
+              return purse;
+            }
+
+            const addressToUpdater = denomToAddressUpdater.get(
+              assetRecord.denom,
+            );
+
+            /** @type {NotifierRecord<Amount>} */
+            const { updater, notifier } = makeNotifierKit();
+            /** @type {bigint} */
+            let lastBalanceUpdate = -1n;
+            /** @type {BalanceUpdater} */
+            const balanceUpdater = (value, nonce = undefined) => {
+              if (nonce !== undefined) {
+                const thisBalanceUpdate = BigInt(nonce);
+                if (thisBalanceUpdate <= lastBalanceUpdate) {
+                  return;
+                }
+                lastBalanceUpdate = thisBalanceUpdate;
+              }
+              // Convert the string value to a bigint.
+              const amt = amountMath.make(brand, BigInt(value));
```

Comment:
This should be Nat rather than BigInt.  balances can never be negative.

Reply:
`Nat` doesn't accept strings.

Reply:
> `Nat` doesn't accept strings.

The `amountMath.make` asserts the non-negative stuff.

Reply:
You can do `Nat(BigInt(value))`